pool/src/pool/actions.rs

```diff
@@ -316,6 +353,38 @@ pub fn build_actions_from_request(
     (actions, from_state, check_health)
 }
 
+fn build_borrow_generic(
+    e: &Env,
+    pool: &mut Pool,
+    from_state: &mut User,
+    actions: &mut Actions,
+    check_health: &mut bool,
+    request: Request,
+    flash: bool,
+) -> i128 {
+    let transfer_amount = request.amount;
+    let token_address = &request.address;
+    let mut reserve = pool.load_reserve(e, token_address, true);
+    let d_tokens_minted = reserve.to_d_token_up(transfer_amount);
+    from_state.add_liabilities(e, &mut reserve, d_tokens_minted);
+    reserve.require_utilization_below_max(e);
+    
+    // the actual difference between the flash borrow and standard borrow.
+    if !flash {
+        actions.add_for_pool_transfer(token_address, transfer_amount);
+    } else {
+        actions.add_flash_borrow(token_address, transfer_amount);
+    }
```

Comment:
It might be clearer to have the individualized code within the switch statement instead of in this function